packages/flutter/lib/src/material/text_field.dart

```diff
@@ -447,6 +448,17 @@ class TextField extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.autofocus}
   final bool autofocus;
 
+  /// Represents the interactive "state" of this widget in terms of
+  /// a set of [MaterialState]s, including [MaterialState.disabled], [MaterialState.hovered],
+  /// [MaterialState.error], and [MaterialState.focused].
+  ///
+  /// Classes based on this one can provide their own
+  /// [MaterialStatesController] to which they've added listeners.
+  /// They can also update the controller's [MaterialStatesController.value]
+  /// however, this may only be done when it's safe to call
+  /// [State.setState], like in an event handler.
```

Comment:
(what's the difference between this one and the existing doc template for this in `ink_well.dart`?)

Reply:
This one just makes sure to point out the states reported by `TextField`. `{ MaterialState.disabled, MaterialState.hovered, MaterialState.error, MaterialState.focused }`. `ink_well.dart` points out `MaterialState.pressed` which `TextField` does not report at the moment.

Reply:
nit: consider linking the part of the material state controller doc that explains that this is not the intrinsic states and if you change the value, future intrinsic states change may overwrite the change?(